--- src/gflags_reporting.cc
+++ src/gflags_reporting.cc
@@ -118,7 +118,7 @@
 // Goes to some trouble to make pretty line breaks.
 string DescribeOneFlag(const CommandLineFlagInfo& flag) {
   string main_part;
-  SStringPrintf(&main_part, "    -%s (%s)",
+  SStringPrintf(&main_part, "    --%s (%s)",
                 flag.name.c_str(),
                 flag.description.c_str());
   const char* c_string = main_part.c_str();
