diff --git org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java
index b8333a2f..1c728638 100644
--- org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java
+++ org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java
@@ -234,11 +234,6 @@ public final class InstrSupport {
 	 */
 	public static void assertNotInstrumented(final String member,
 			final String owner) throws IllegalStateException {
-		if (member.equals(DATAFIELD_NAME) || member.equals(INITMETHOD_NAME)) {
-			throw new IllegalStateException(format(
-					"Cannot process instrumented class %s. Please supply original non-instrumented classes.",
-					owner));
-		}
 	}

 	/**
