diff --git compiler-rt/lib/fuzzer/FuzzerCommand.h compiler-rt/lib/fuzzer/FuzzerCommand.h
index 87308864af53..145063d000fb 100644
--- compiler-rt/lib/fuzzer/FuzzerCommand.h
+++ compiler-rt/lib/fuzzer/FuzzerCommand.h
@@ -140,7 +140,7 @@ public:
   std::string toString() const {
     std::stringstream SS;
     for (auto arg : getArguments())
-      SS << arg << " ";
+      SS << "\"" << arg << "\" ";
     if (hasOutputFile())
       SS << ">" << getOutputFile() << " ";
     if (isOutAndErrCombined())
